tests/RobotFramework/tests/cumulocity/self-update/tedge_self_update.robot

```diff
@@ -47,6 +49,34 @@ Update tedge version from previous using Cumulocity
     ${OUTPUT}    Execute Command    systemctl is-enabled tedge-mapper-az || exit 1    exp_exit_code=1    strip=True
     Should Be Equal    ${OUTPUT}    disabled    msg=Service should still be disabled
 
+
+Refreshes mosquitto bridge configuration
+    ${PREV_VERSION}=    Set Variable    0.10.0
+    # Install base version
+    Execute Command    curl -fsSL https://raw.githubusercontent.com/thin-edge/thin-edge.io/main/get-thin-edge_io.sh | sudo sh -s ${PREV_VERSION}
+
+    # Register device (using already installed version)
+    Execute Command    cmd=test -f ./bootstrap.sh && env DEVICE_ID=${DEVICE_SN} ./bootstrap.sh --no-install --no-secure || true
+    Device Should Exist                      ${DEVICE_SN}
+
+    # get bridge modification time
+    ${before_upgrade_epoch_str}=    Execute Command      stat /etc/tedge/mosquitto-conf/c8y-bridge.conf -c %Y    strip=True
+    ${before_upgrade_epoch}=        Convert To Number    ${before_upgrade_epoch_str}
+    ${before_upgrade_time}=         Convert Date         ${before_upgrade_epoch}
+
+    # Install newer version
+    Create Local Repository
+    ${OPERATION}=    Install Software    tedge,${NEW_VERSION}    tedge-mapper,${NEW_VERSION}    tedge-agent,${NEW_VERSION}    tedge-watchdog,${NEW_VERSION}    tedge-apt-plugin,${NEW_VERSION}
+    Operation Should Be SUCCESSFUL    ${OPERATION}    timeout=180
+
+    # TODO: check that this new configuration is actually used by mosquitto
+    ${c8y_bridge_mod_epoch_str}=    Execute Command      stat /etc/tedge/mosquitto-conf/c8y-bridge.conf -c %Y    strip=True
+    ${c8y_bridge_mod_epoch}=    Convert To Number    ${c8y_bridge_mod_epoch_str}
+    ${c8y_bridge_mod_time}=    Convert Date         ${c8y_bridge_mod_epoch}
+
+    ${c8y_bridge_mod_time_after_upgrade}=         Subtract Date From Date    ${c8y_bridge_mod_time}    ${before_upgrade_time}
+    Should Be True                                ${c8y_bridge_mod_time_after_upgrade} > 0
```

Comment:
This is definitely accurate, but just checking if `c8y_bridge_mod_epoch_str != before_upgrade_epoch_str` would have been sufficient and simpler.

Reply:
fixed